stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/composer/attachment/preview/factory/MediaAttachmentPreviewFactory.kt

```diff
@@ -0,0 +1,152 @@
+                    binding = binding,
+                    attachmentRemovalListener = attachmentRemovalListener,
+                    style = style
+                )
+            }
+    }
+
+    /**
+     * A ViewHolder for image and video attachment previews.
+     *
+     * @param binding Binding generated for the layout.
+     * @param attachmentRemovalListener Click listener for the remove attachment button.
+     */
+    private class MediaAttachmentPreviewViewHolder(
+        private val binding: StreamUiMediaAttachmentPreviewBinding,
+        attachmentRemovalListener: (Attachment) -> Unit,
+        private val style: MessageComposerViewStyle?,
+    ) : AttachmentPreviewViewHolder(binding.root) {
+
+        private lateinit var attachment: Attachment
+
+        init {
+            val cornerRadius = context.getDimension(R.dimen.stream_ui_selected_attachment_corner_radius)
+                .toFloat()
+            binding.thumbImageView.shapeAppearanceModel = ShapeAppearanceModel.builder()
+                .setAllCornerSizes(cornerRadius)
+                .build()
+            binding.removeButton.setOnClickListener { attachmentRemovalListener(attachment) }
+            setupIconImageView()
+            setupIconCard()
+        }
+
+        /**
+         * Applies the style to the View displaying the play button
+         * icon.
+         */
+        private fun setupIconImageView() {
+            if (style != null) {
+                with(binding.playIconImageView) {
+                    val iconDrawable =
+                        style.messageInputVideoAttachmentIconDrawable.applyTint(
+                            style.messageInputVideoAttachmentIconDrawableTint
+                        )
+
+                    load(iconDrawable)
+                    setPaddingRelative(
+                        style.messageInputVideoAttachmentIconDrawablePaddingStart,
+                        style.messageInputVideoAttachmentIconDrawablePaddingTop,
+                        style.messageInputVideoAttachmentIconDrawablePaddingEnd,
+                        style.messageInputVideoAttachmentIconDrawablePaddingBottom
+                    )
+                }
+            }
+        }
+
+        /**
+         * Applies style to the card holding the View displaying the
+         * play button.
+         */
+        private fun setupIconCard() {
+            if (style != null) {
+                with(binding.playIconCardView) {
+                    style.messageInputVideoAttachmentIconBackgroundColor?.also { backgroundColor ->
+                        setCardBackgroundColor(backgroundColor)
+                    }
+                    elevation = style.messageInputVideoAttachmentIconElevation
+                }
+            }
+        }
+
+        override fun bind(attachment: Attachment) {
+            this.attachment = attachment
+            val upload = attachment.upload
+
+            binding.playIconCardView.isVisible = attachment.isVideo()
```

Comment:
I think we don't need to hide it's child as well separately since parent visibility propagates, but if someone remembers and Android gotcha please say so. :smile: